### PR TITLE
MSC3939: Account locking

### PR DESCRIPTION
**Pull request by Johennes**
Spec for [MSC3939](https://github.com/matrix-org/matrix-spec-proposals/pull/3939).

### Pull Request Checklist

* [x] Pull request includes a [changelog file](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#adding-to-the-changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#sign-off)
* [ ] Pull request is classified as ['other changes'](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#other-changes)

Preview: https://pr1934--matrix-spec-previews.netlify.app

### CODE REVIEW

File: changelogs/client_server/newsfragments/1934.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Add account locking as per MSC3939.
```
Comment (Johennes) [inline changelogs/client_server/newsfragments/1934.feature:1]:
```suggestion
Add account locking as per [MSC3939](https://github.com/matrix-org/matrix-spec-proposals/pull/MSC3939).
```

File: changelogs/client_server/newsfragments/1934.feature [hunk 2]
```diff
@@ -0,0 +1 @@
+Add account locking as per [MSC3939](https://github.com/matrix-org/matrix-spec-proposals/pull/MSC3939).
```
Comment (Johennes) [inline changelogs/client_server/newsfragments/1934.feature:1]:
```suggestion
Add account locking as per [MSC3939](https://github.com/matrix-org/matrix-spec-proposals/pull/3939).
```

File: content/client-server-api/_index.md [hunk 3]
```diff
@@ -1425,6 +1432,55 @@ number and a symbol and be at a minimum 8 characters in length. Servers
...
+Servers SHOULD NOT invalidate access tokens on locked accounts unless the
+client requests a logout (using the above endpoints). This ensures that 
+users can retain their sessions without having to log back in if the account
+becomes unlocked.
+
+Upon receiving an `M_USER_LOCKED` error, clients SHOULD retain session
+information including encryption state and inform the user that their account
+has been locked. While the lock is applied, clients SHOULD hide the normal UI
+from the user, preventing general use of their account. Clients SHOULD, however,
+continue to make rate-limited requests to `/sync` and other APIs to detect
```
[the remark this answers is not in the document]
Comment (Johennes) [reply content/client-server-api/_index.md]: Good idea. 👍

**Review by anoadragon453 — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**